core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -1026,6 +1026,21 @@ public boolean isDetached() {
         return DetachedPluginsUtil.isDetachedPlugin(shortName);
     }
 
+    @Restricted(NoExternalUse.class)
+    public boolean hasImpliedDependents() {
+        if (!isDetached()) {
+            return false;
+        }
+        for (PluginWrapper p : Jenkins.get().getPluginManager().getPlugins()) {
```

Comment:
Do you not need to check that `p.enabled`?

Reply:
I thought about it, but didn't for two reasons:

* We're currently not live-updating the list when the enabled status of entries is changed. Disable all possible dependents and the message still exists; same with enabling without reload.
* I didn't think it was worth it to figure out what happens when you re-enable a disabled plugin. IIRC this can even happen in a running instance when you install a plugin with a dependency on the disabled plugin.

All of this seemed like too much effort for too little gain (not showing this message in few cases).